Flybys activity

jeffschuler opened the pull request:
Hi, and thanks much for creating this plugin!

I was wondering whether you'd entertain adding a functionality:

On the Flybys Viewer (in Strava Labs) by default all Flyby activities (whether runs or rides) are shown. I often check Flybys to see who I passed on the way and, as a runner, am almost always interested only in other runners rather than cyclists.

This patch allows auto-clicking of the "Runs only" (or "Rides only"; depends on the activity) checkbox, obviously adding an option for this to the plugin options page.

Also, because the "Athlete Profile" function throws a JS error on pages where there is no user menu, I had to add a check in there, otherwise the plugin fails on pages like Strava Labs.

Thanks for your consideration!

Jeff

lamby commented on js/main.js:
```diff
@@ -844,7 +844,11 @@ function StravaEnhancementSuite($, options) {
   // Athlete profile
   $.always(function() {
-    if ($('header .user-menu a').attr('href').indexOf(window.location.pathname) !== 0) {
+    profile_link = $('header .user-menu a');
```
Missing ``var`` here?

lamby commented on js/main.js:
```diff
@@ -1283,6 +1283,20 @@ function StravaEnhancementSuite($, options) {
+    $(document).ready(function() {
+      // Check the "Runs only" / "Rides only" checkbox
+      var elem = $('input#hide_different_activity_type');
+      if (elem.checked !== 'on') {
```
Should we be using ``elem.prop('checked')`` here instead?

lamby commented:
Looks great, very very happy to merge if you could fix up and/or address these two small niggles..?

jeffschuler commented:
Yes, thanks!

Committed.

I'm seeing some inconsistency, though, with this feature. Sometimes it works fine, but sometimes the Flybys list isn't loaded yet and the checkbox gets clicked but without the desired effect. I'll have to try something different than $(document).ready().

lamby commented:
> I'll have to try something different than $(document).ready().

Ah yes. So, I think you can just write this as:

```
// Show same-activity Flybys only (runs or rides) in the Flyby viewer.
$.option('show_same_activity_flybys', function() {
  if (window.location.pathname.indexOf('/flyby/viewer') !== 0) {
    return;
  }

  $.setInterval(function() {
    // Check the "Runs only" / "Rides only" checkbox
    $('input#hide_different_activity_type')
      .onceOnly()
      .click()
      ;
  }, 1000);
});

```

jeffschuler commented:
I saw that... I'm not crazy about the fact that it continues to run every second. I see there are a number of these running continually now when I browse Strava pages with the plugin.

I dug through Strava's JS a bit to try to find an event or callback to latch onto, but haven't had success yet.

lamby commented:
>  not crazy about the fact that it continues to run every second

Agreed. However, I have plans to fix all of these at once, so if you simply follow the existing pattern then it will be easy to fix them all in one go.

jeffschuler commented:
Got it. And glad to hear.

Unfortunately, that approach alone still suffers the same issue where the button is clicked before it can have an effect.

I've switched over to using onceOnly() and added an extra check to ensure the activities table is loaded before clicking.

lamby commented:
Thanks!

lamby commented:
Released as 2.7.0

jeffschuler commented:
Thank you!
